x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/expression/function/SqlFunctionRegistryTests.java

```diff
@@ -0,0 +1,130 @@
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.sql.expression.function;
+
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.xpack.ql.ParsingException;
+import org.elasticsearch.xpack.ql.expression.Expression;
+import org.elasticsearch.xpack.ql.expression.function.FunctionDefinition;
+import org.elasticsearch.xpack.ql.expression.function.FunctionRegistry;
+import org.elasticsearch.xpack.ql.expression.function.FunctionRegistryTests;
+import org.elasticsearch.xpack.ql.expression.function.FunctionRegistryTests.DummyFunction;
+import org.elasticsearch.xpack.ql.expression.function.UnresolvedFunction;
+import org.elasticsearch.xpack.ql.session.Configuration;
+import org.elasticsearch.xpack.ql.tree.Source;
+
+import java.time.ZoneId;
+
+import static org.elasticsearch.xpack.ql.TestUtils.randomConfiguration;
+import static org.elasticsearch.xpack.ql.expression.function.FunctionRegistry.def;
+import static org.elasticsearch.xpack.ql.expression.function.FunctionRegistryTests.uf;
+import static org.elasticsearch.xpack.ql.expression.function.FunctionResolutionStrategy.DEFAULT;
+import static org.elasticsearch.xpack.sql.expression.function.SqlFunctionResolution.DISTINCT;
+import static org.elasticsearch.xpack.sql.expression.function.SqlFunctionResolution.EXTRACT;
+import static org.hamcrest.Matchers.endsWith;
+import static org.mockito.Mockito.mock;
+
+public class SqlFunctionRegistryTests extends ESTestCase {
+
+    public void testNoArgFunction() {
+        UnresolvedFunction ur = uf(DEFAULT);
+        FunctionRegistry r = new FunctionRegistry(def(DummyFunction.class, DummyFunction::new, "DUMMY_FUNCTION"));
+        FunctionDefinition def = r.resolveFunction(ur.name());
+        assertEquals(ur.source(), ur.buildResolved(randomConfiguration(), def).source());
+
+        // Distinct isn't supported
+        ParsingException e = expectThrows(ParsingException.class, () ->
+            uf(DISTINCT).buildResolved(randomConfiguration(), def));
+        assertThat(e.getMessage(), endsWith("does not support DISTINCT yet it was specified"));
+
+        // Any children aren't supported
+        e = expectThrows(ParsingException.class, () ->
+            uf(DEFAULT, mock(Expression.class)).buildResolved(randomConfiguration(), def));
+        assertThat(e.getMessage(), endsWith("expects no arguments"));
+    }
+
+    public void testUnaryFunction() {
+        UnresolvedFunction ur = uf(DEFAULT, mock(Expression.class));
+        FunctionRegistry r = new FunctionRegistry(def(DummyFunction.class, (Source l, Expression e) -> {
+            assertSame(e, ur.children().get(0));
+            return new DummyFunction(l);
+        }, "DUMMY_FUNCTION"));
+        FunctionDefinition def = r.resolveFunction(ur.name());
+        assertFalse(def.extractViable());
+        assertEquals(ur.source(), ur.buildResolved(randomConfiguration(), def).source());
```

Comment:
This test is a duplicate of a test inside `FunctionRegistryTests` in QL.

Reply:
`FunctionRegistryTests`:

```
    public void testNoArgFunction() {
        UnresolvedFunction ur = uf(DEFAULT);
        FunctionRegistry r = new FunctionRegistry(def(DummyFunction.class, DummyFunction::new, "DUMMY_FUNCTION"));
        FunctionDefinition def = r.resolveFunction(ur.name());
        assertEquals(ur.source(), ur.buildResolved(randomConfiguration(), def).source());
    }
```

`SqlFunctionRegistryTests`:

```
    public void testNoArgFunction() {
        UnresolvedFunction ur = uf(DEFAULT);
        FunctionRegistry r = new FunctionRegistry(def(DummyFunction.class, DummyFunction::new, "DUMMY_FUNCTION"));
        FunctionDefinition def = r.resolveFunction(ur.name());
        assertEquals(ur.source(), ur.buildResolved(randomConfiguration(), def).source());

        // Distinct isn't supported
        ParsingException e = expectThrows(ParsingException.class, () ->
            uf(DISTINCT).buildResolved(randomConfiguration(), def));
        assertThat(e.getMessage(), endsWith("does not support DISTINCT yet it was specified"));

        // Any children aren't supported
        e = expectThrows(ParsingException.class, () ->
            uf(DEFAULT, mock(Expression.class)).buildResolved(randomConfiguration(), def));
        assertThat(e.getMessage(), endsWith("expects no arguments"));
    }
```